From 535cb44a876bd106bc4b8d98a4d873fec5b4f0c5 Mon Sep 17 00:00:00 2001
From: "OKHand.ziyu" <OKHand.ziyu@gmail.com>
Date: Fri, 22 Nov 2024 10:10:17 +0800
Subject: [PATCH 11/26] filter name miss

---
 exo/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exo/main.py b/exo/main.py
index 8dd51e7..b6cf770 100644
--- a/exo/main.py
+++ b/exo/main.py
@@ -239,7 +239,7 @@ async def run_model_cli(node: Node, inference_engine: InferenceEngine, model_nam
       print(f"Error: Not Find Local model '{model_name}' for inference engine {inference_engine.__class__.__name__}")
       return
   else:
-    print(f"Error: Unsupported model '{model_name}' for inference engine {inference_engine.__class__.__name__}")
+    print(f"Error: Unsupported model '{model_name_or_path}' for inference engine {inference_engine.__class__.__name__}")
     return
   
   request_id = str(uuid.uuid4())
-- 
2.39.5 (Apple Git-154)

